core/trino-main/src/main/java/io/trino/operator/output/PagePartitioner.java

```diff
@@ -146,11 +148,51 @@ public void partitionPage(Page page, OperatorContext operatorContext)
         else {
             partitionPageByColumn(page);
         }
-        long positionsAppendersSizeAfter = getPositionsAppendersSizeInBytes();
-        flushPositionsAppenders(false);
+        long outputSizeInBytes = flushPositionsAppenders(false);
         updateMemoryUsage();
 
-        operatorContext.recordOutput(positionsAppendersSizeAfter - positionsAppendersSizeBefore, outputPositionCount);
+        recordOutputStats(operatorContext, outputSizeInBytes, outputPositionCount);
+    }
+
+    private void recordOutputStats(OperatorContext operatorContext, long outputSizeInBytes, int outputPositions)
+    {
+        // The outputSizeInBytes already reported during release
+        if (outputSizeReportedBeforeRelease > 0) {
+            long consumeFromPreviouslyReported = min(outputSizeReportedBeforeRelease, outputSizeInBytes);
```

Comment:
I must say I don't understand this logic. Could we make it more straight-forward? Why `outputSizeInBytes` would be smaller than `outputSizeReportedBeforeRelease`?

Reply:
It shouldn't but this is defensive code that ensures that we never allow the "eagerly reported" quantity to go negative.